### DISCUSSION TITLE
Adding multiple AWS Credentials in Vector Helm chart Deployment

### DISCUSSION THREAD

**Discussion by garceger** [vectordotdev/vector#20623] [Q&A]
Hello everyone,

We're currently deploying Vector using the Helm chart and injecting AWS credentials via environment variables. However, we need to send logs to an AWS S3 bucket **located in a different AWS account**. We've tried creating an IAM Role pointing to the other account's S3 bucket, but it hasn't worked as expected as the pod couldn't assume the correct role. Additionally, we're unsure how to handle these additional credentials within the Helm chart.

We've also attempted to create a separate file containing the other credentials, but we're unsure of the correct approach for integrating it via the Helm chart.

Any guidance or suggestions would be greatly appreciated!

Thanks,
Germán

**Comment by jszwedko** [comment]
I think you can create an IAM role in the account that the pod is deployed in and then, in the account the S3 bucket is in, use a bucket policy to grant access to the role to write. https://repost.aws/knowledge-center/cross-account-access-s3 might be helpful.
